### ISSUE TITLE
Recording performances

### ISSUE THREAD

**Issue by Tronic** [performous/performous#2]
Record webcam video (if available), audio and input event information to be stored with a high score, or allow sharing online.

There are a number of technical difficulties that need to be solved.
-Syncinc audio clocks together for a single audio track that matches the original song
-A/V compression with ffmpeg (realtime in a separate thread?)
-Online features (separate issue)
-Input storage format for instruments and dance (if supported at all)
-Disk storage format (is video stored only once and each player stored separately?)

**Comment by Baklap4** [comment]
There's an experimental flag to save webcam video feed to a file https://github.com/performous/performous/blob/master/game/webcam.cc#L47-L58 not quite sure if it covers all or nothing at all

**Comment by Tronic** [comment]
Nice to see someone working on this. However, notice that you need to get audio time to timestamp video frames correctly because webcams do not actually run at a steady frame rate and writing a 30 FPS video will quickly get out of sync with the song.

Probably you need to use ffmpeg for video writing because OpenCV is *very* limited.
